lib/httpPipelineLogger.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { HttpPipelineLogLevel } from "./httpPipelineLogLevel";
+
+/**
+ * A Logger that can be added to a HttpPipeline. This enables each RequestPolicy to log messages
+ * that can be used for debugging purposes.
+ */
+export interface HttpPipelineLogger {
+  /**
+   * The log level threshold for what logs will be logged.
+   * @return The log level threshold for what logs will be logged.
+   */
+  minimumLogLevel: HttpPipelineLogLevel;
+
+  /**
+   * Log the provided message.
+   * @param logLevel The HttpLogDetailLevel associated with this message.
+   * @param message The message to log.
+   * @param formattedArguments A variadic list of arguments that should be formatted into the
+   *                           provided message.
+   */
+  log(logLevel: HttpPipelineLogLevel, message: string): void;
+}
+
+/**
+ * A HttpPipelineLogger that will send its logs to the console.
+ */
+export class ConsoleHttpPipelineLogger implements HttpPipelineLogger {
+  /**
+   * Create a new ConsoleHttpPipelineLogger.
+   * @param minimumLogLevel The log level threshold for what logs will be logged.
+   */
+  constructor(public minimumLogLevel: HttpPipelineLogLevel) {
+  }
+
+  /**
+   * Log the provided message.
+   * @param logLevel The HttpLogDetailLevel associated with this message.
+   * @param message The message to log.
+   * @param formattedArguments A variadic list of arguments that should be formatted into the
+   *                           provided message.
+   */
+  log(logLevel: HttpPipelineLogLevel, message: string): void {
+    console.log(`${HttpPipelineLogLevel[logLevel]}: ${message}`);
```

Comment:
This is not a blocker or anything, but consider using the console.warn, console.error etc variants depending on the log level

Reply:
That's a good idea. I'll do that.

Reply:
Fixed in https://github.com/Azure/ms-rest-js/pull/76/commits/7d36607f46420aae08b8d7824b13398007bbdcc0.